## 1. This patient may have sepsis
<p>Sepsis is a deadly syndrome where a patient has a severe infection that causes organ failure. The sooner septic patients are treated, the more likely they are to survive, but sepsis can be challenging to recognize. It may be possible to use hospital data to develop machine learning models that could flag patients who are likely to be septic. However, before we develop predictive algorithms, we need a reliable method to determine patients who are septic. One component of sepsis is a severe infection.</p>
<p>In this project, we will use two weeks of hospital electronic health record (EHR) data to find out which patients had a severe infection according to four criteria. We will look into the data to see if a doctor ordered a blood test to look for bacteria (a blood culture) and gave the patient a series of intervenous antibiotics. </p>
<p>Let's get started!</p>

In [23]:
# Load packages
library(data.table)
antibioticDT <- fread("antibioticDT.csv")


# Look at the first 30 rows

head(antibioticDT,30)

# .... YOUR CODE FOR TASK 1 ....

patient_id day_given antibiotic_type route
1   1          2        ciprofloxacin   IV   
2   1          4        ciprofloxacin   IV   
3   1          6        ciprofloxacin   IV   
4   1          7        doxycycline     IV   
5   1          9        doxycycline     IV   
6   1         15        penicillin      IV   
7   1         16        doxycycline     IV   
8   1         18        ciprofloxacin   IV   
9   8          1        doxycycline     PO   
10  8          2        penicillin      IV   
11  8          3        doxycycline     IV   
12  8          6        doxycycline     PO   
13  8          8        penicillin      PO   
14  8         12        penicillin      IV   
15  9          8        doxycycline     IV   
16  9         12        doxycycline     PO   
17 12          4        doxycycline     PO   
18 12          9        doxycycline     IV   
19 16          1        doxycycline     IV   
20 16          4        amoxicillin     IV   
21 19          3        doxycycline     PO   
22 19          5        amoxicillin     IV   
23 19          6        ciprofloxacin   IV   
24 19         10        doxycycline     IV   
25 19         12        penicillin      IV   
26 23          1        doxycycline     IV   
27 23          1        penicillin      IV   
28 23          3        amoxicillin     IV   
29 23          3        ciprofloxacin   IV   
30 23          3        doxycycline     IV

## 2. Which antibiotics are "new"?
<p>These data represent all drugs administered in a hospital over two weeks. Each row represents one time a patient was given an antibiotic. The variables include the patient identification number, the day the drug was administered, the name of the antibiotic, and how it was administered. For example, patient "8" received doxycycline by mouth on the first day of their stay.</p>
<p>We will identify patients with a serious infection using the following criteria. </p>
<p><strong>Criteria for Suspected Infection</strong><a href="https://www.ncbi.nlm.nih.gov/pubmed/28903154">*</a></p>
<ol>
<li>The patient receives antibiotics for a sequence of four days, with gaps of one day allowed.</li>
<li>The sequence must start with a new antibiotic, defined as an antibiotic type that was not given in the previous two days.</li>
<li>The sequence must start within two days of a blood culture.</li>
<li>There must be at least one intervenous (I.V.) antibiotic within the +/-2 day window.</li>
</ol>
<p>Let's start with the second item by finding which rows represent "new antibiotics". We will determine if each antibiotic was given to the patient in the prior two days. We'll visualize this task by looking at the data sorted by id, antibiotic type, and day.</p>

In [24]:
# Sort the data and examine the first 40 rows
# .... YOUR CODE FOR TASK 2 .....

setorder(antibioticDT, patient_id, antibiotic_type, day_given)
#head(antibioticDT, 30)

# Use shift to calculate the last day a particular drug was administered

antibioticDT[ , last_administration_day := shift(day_given, n=1, type = "lag", fill=NA), by = .(patient_id, antibiotic_type)]
#head(antibioticDT, 30)

# Calculate the number of days since the drug was last administered

antibioticDT[ , days_since_last_admin := (day_given - last_administration_day), by=.(patient_id)]
#head(antibioticDT, 30)

# Create antibiotic_new with an initial value of one, then reset it to zero as needed
# .... YOUR CODE FOR TASK 2 .....


antibioticDT[, antibiotic_new:=1]

antibioticDT[days_since_last_admin <= 2, antibiotic_new := 0]

head(antibioticDT, 30)

patient_id day_given antibiotic_type route last_administration_day
1   1          2        ciprofloxacin   IV    NA                     
2   1          4        ciprofloxacin   IV     2                     
3   1          6        ciprofloxacin   IV     4                     
4   1         18        ciprofloxacin   IV     6                     
5   1          7        doxycycline     IV    NA                     
6   1          9        doxycycline     IV     7                     
7   1         16        doxycycline     IV     9                     
8   1         15        penicillin      IV    NA                     
9   8          1        doxycycline     PO    NA                     
10  8          3        doxycycline     IV     1                     
11  8          6        doxycycline     PO     3                     
12  8          2        penicillin      IV    NA                     
13  8          8        penicillin      PO     2                     
14  8         12        penicillin      IV     8                     
15  9          8        doxycycline     IV    NA                     
16  9         12        doxycycline     PO     8                     
17 12          4        doxycycline     PO    NA                     
18 12          9        doxycycline     IV     4                     
19 16          4        amoxicillin     IV    NA                     
20 16          1        doxycycline     IV    NA                     
21 19          5        amoxicillin     IV    NA                     
22 19          6        ciprofloxacin   IV    NA                     
23 19          3        doxycycline     PO    NA                     
24 19         10        doxycycline     IV     3                     
25 19         12        penicillin      IV    NA                     
26 23          3        amoxicillin     IV    NA                     
27 23          8        amoxicillin     IV     3                     
28 23         10        amoxicillin     PO     8                     
29 23          3        ciprofloxacin   IV    NA                     
30 23          5        ciprofloxacin   PO     3                     
   days_since_last_admin antibiotic_new
1  NA                    1             
2   2                    0             
3   2                    0             
4  12                    1             
5  NA                    1             
6   2                    0             
7   7                    1             
8  NA                    1             
9  NA                    1             
10  2                    0             
11  3                    1             
12 NA                    1             
13  6                    1             
14  4                    1             
15 NA                    1             
16  4                    1             
17 NA                    1             
18  5                    1             
19 NA                    1             
20 NA                    1             
21 NA                    1             
22 NA                    1             
23 NA                    1             
24  7                    1             
25 NA                    1             
26 NA                    1             
27  5                    1             
28  2                    0             
29 NA                    1             
30  2                    0

### Observation from sorting the data using setorder():

-> patient_id is still arranged in ascending order.

-> day_given variable has been arranged serially according to the alphabetial order in antibiotic_type variable due to sorting the data on antibiotic_type variable.

-> antibiotic_type is now arranged alphabetically for each patient

## 3. Looking at the blood culture data
<p>Now let's look at blood culture data from the same two-week period in this hospital. These data are in <code>blood_cultureDT.csv</code>. Let's start by reading it into the workspace and having a look at a few rows. </p>
<p>Each row represents one blood culture and gives the patient's id and the day the blood culture test occurred. For example, patient "8" had a blood culture on the second day of their hospitalization and again on the thirteenth day. Notice that some patients from the antibiotic dataset are not in this dataset and vice versa. Some patients are in neither because they received neither antibiotics nor a blood culture.</p>

In [25]:
# Read in blood_cultureDT.csv
blood_cultureDT <- fread('blood_cultureDT.csv')
head(blood_cultureDT,30)

# Print the first 30 rows
# .... YOUR CODE FOR TASK 3 ....

patient_id blood_culture_day
1    1         3               
2    1        13               
3    8         2               
4    8        13               
5   23         3               
6   39        10               
7   45         4               
8   45         9               
9   45        11               
10  51         3               
11  51         6               
12  59         2               
13  64         1               
14  66         9               
15  66        10               
16  69         2               
17  69         6               
18  69         7               
19  69        11               
20  69        16               
21  76         1               
22  77         3               
23  79         5               
24  79        11               
25  79        12               
26  80         3               
27  80        12               
28  81         2               
29 112         6               
30 115         2

## 4. Combine the antibiotic data and the blood culture data
<p>To find which antibiotics were given close to a blood culture test, we need to combine the drug administration data with the blood culture data. We'll keep only patients that are still candidates for infection&mdash;only those in both data sets.</p>
<p>A challenge with the data is that some patients had blood cultures on several different days. For each of those days, we will see if there is a sequence of antibiotic days close to them. To accomplish this, in the merge we will match each blood culture to each antibiotic day.</p>
<p>After sorting the data following the merge, you will see that each patient's antibiotic sequence repeats for each blood culture day. This repetition allows us to look at each blood culture day and check if it is associated with a qualifying sequence of antibiotics.</p>

In [26]:
# Merge antibioticDT with blood_cultureDT
combinedDT <- merge.data.table(antibioticDT, blood_cultureDT, by = 'patient_id', all = FALSE)

# Sort by patient_id, blood_culture_day, day_given, and antibiotic_type
# .... YOUR CODE FOR TASK 4 .....

setorder(combinedDT, patient_id, blood_culture_day, day_given, antibiotic_type)
# Print and examine the first 30 rows
# .... YOUR CODE FOR TASK 4 .....

head(combinedDT,30)

patient_id day_given antibiotic_type route last_administration_day
1   1          2        ciprofloxacin   IV    NA                     
2   1          4        ciprofloxacin   IV     2                     
3   1          6        ciprofloxacin   IV     4                     
4   1          7        doxycycline     IV    NA                     
5   1          9        doxycycline     IV     7                     
6   1         15        penicillin      IV    NA                     
7   1         16        doxycycline     IV     9                     
8   1         18        ciprofloxacin   IV     6                     
9   1          2        ciprofloxacin   IV    NA                     
10  1          4        ciprofloxacin   IV     2                     
11  1          6        ciprofloxacin   IV     4                     
12  1          7        doxycycline     IV    NA                     
13  1          9        doxycycline     IV     7                     
14  1         15        penicillin      IV    NA                     
15  1         16        doxycycline     IV     9                     
16  1         18        ciprofloxacin   IV     6                     
17  8          1        doxycycline     PO    NA                     
18  8          2        penicillin      IV    NA                     
19  8          3        doxycycline     IV     1                     
20  8          6        doxycycline     PO     3                     
21  8          8        penicillin      PO     2                     
22  8         12        penicillin      IV     8                     
23  8          1        doxycycline     PO    NA                     
24  8          2        penicillin      IV    NA                     
25  8          3        doxycycline     IV     1                     
26  8          6        doxycycline     PO     3                     
27  8          8        penicillin      PO     2                     
28  8         12        penicillin      IV     8                     
29 23          1        doxycycline     IV    NA                     
30 23          1        penicillin      IV    NA                     
   days_since_last_admin antibiotic_new blood_culture_day
1  NA                    1               3               
2   2                    0               3               
3   2                    0               3               
4  NA                    1               3               
5   2                    0               3               
6  NA                    1               3               
7   7                    1               3               
8  12                    1               3               
9  NA                    1              13               
10  2                    0              13               
11  2                    0              13               
12 NA                    1              13               
13  2                    0              13               
14 NA                    1              13               
15  7                    1              13               
16 12                    1              13               
17 NA                    1               2               
18 NA                    1               2               
19  2                    0               2               
20  3                    1               2               
21  6                    1               2               
22  4                    1               2               
23 NA                    1              13               
24 NA                    1              13               
25  2                    0              13               
26  3                    1              13               
27  6                    1              13               
28  4                    1              13               
29 NA                    1               3               
30 NA                    1               3

### Observation from above operation:

A slight change we notice in the table above is that for each blood_culture_day, we get the data repeated twice

## 5. Determine whether each row is in-window
<p>Now that we have the antibiotic and blood culture data combined, we can test each drug administration against each blood culture to see if it's "in the window."</p>

In [27]:
# Make a new variable called drug_in_bcx_window
# ....YOUR CODE FOR TASK 5 ....

combinedDT[, drug_in_bcx_window := as.numeric(abs(blood_culture_day - day_given) <= 2)]

head(combinedDT)

patient_id day_given antibiotic_type route last_administration_day
1 1           2        ciprofloxacin   IV    NA                     
2 1           4        ciprofloxacin   IV     2                     
3 1           6        ciprofloxacin   IV     4                     
4 1           7        doxycycline     IV    NA                     
5 1           9        doxycycline     IV     7                     
6 1          15        penicillin      IV    NA                     
  days_since_last_admin antibiotic_new blood_culture_day drug_in_bcx_window
1 NA                    1              3                 1                 
2  2                    0              3                 1                 
3  2                    0              3                 0                 
4 NA                    1              3                 0                 
5  2                    0              3                 0                 
6 NA                    1              3                 0

### Observations from above output:

The new variable, any_iv_in_bcx_window, indicates whether or not an I.V. drug was given within a +/-2 day window of a blood culture day by 1. And by 0 if both the criterias do not meet

## 6. Check the I.V. requirement
<p>Now let's look at the fourth item in the criteria. </p>
<p><strong>Criteria for Suspected Infection</strong><a href="https://www.ncbi.nlm.nih.gov/pubmed/28903154">*</a></p>
<ol>
<li>The patient receives antibiotics for a sequence of four days, with gaps of one day allowed.</li>
<li>The sequence must start with a new antibiotic, defined as an antibiotic type that was not given in the previous two days.</li>
<li>The sequence must start within two days of a blood culture.</li>
<li><em>There must be at least one intervenous (I.V.) antibiotic within the +/-2 day window.</em></li>
</ol>

In [28]:
# Create a variable indicating if there was at least one I.V. drug given in the window
# .... YOUR CODE FOR TASK 6 ....
combinedDT[, any_iv_in_bcx_window := as.numeric(any(route == "IV" & drug_in_bcx_window == 1)), by = .(blood_culture_day, patient_id)]
head(combinedDT)

# Exclude rows in which the blood_culture_day does not have any I.V. drugs in window 
# .... YOUR CODE FOR TASK 6 ....
combinedDT <- combinedDT[any_iv_in_bcx_window == 1]
head(combinedDT)

patient_id day_given antibiotic_type route last_administration_day
1 1           2        ciprofloxacin   IV    NA                     
2 1           4        ciprofloxacin   IV     2                     
3 1           6        ciprofloxacin   IV     4                     
4 1           7        doxycycline     IV    NA                     
5 1           9        doxycycline     IV     7                     
6 1          15        penicillin      IV    NA                     
  days_since_last_admin antibiotic_new blood_culture_day drug_in_bcx_window
1 NA                    1              3                 1                 
2  2                    0              3                 1                 
3  2                    0              3                 0                 
4 NA                    1              3                 0                 
5  2                    0              3                 0                 
6 NA                    1              3                 0                 
  any_iv_in_bcx_window
1 1                   
2 1                   
3 1                   
4 1                   
5 1                   
6 1

patient_id day_given antibiotic_type route last_administration_day
1 1           2        ciprofloxacin   IV    NA                     
2 1           4        ciprofloxacin   IV     2                     
3 1           6        ciprofloxacin   IV     4                     
4 1           7        doxycycline     IV    NA                     
5 1           9        doxycycline     IV     7                     
6 1          15        penicillin      IV    NA                     
  days_since_last_admin antibiotic_new blood_culture_day drug_in_bcx_window
1 NA                    1              3                 1                 
2  2                    0              3                 1                 
3  2                    0              3                 0                 
4 NA                    1              3                 0                 
5  2                    0              3                 0                 
6 NA                    1              3                 0                 
  any_iv_in_bcx_window
1 1                   
2 1                   
3 1                   
4 1                   
5 1                   
6 1

### Observations from the above operation:

As we want to exculde the rows that don't have IV drug in the window, we already have created a variable for this purpose called any_iv_in_bcx_window. 1 means this criteria is met and 0 means it doesn't meet.
As we want only the the ones having IV drug in the window, we subset the ones with 1 in any_iv_in_bcx_window

## 7. Find the first day of possible sequences
<p>We're getting close! Let's review the criteria again.</p>
<p><strong>Criteria for Suspected Infection</strong><a href="https://www.ncbi.nlm.nih.gov/pubmed/28903154">*</a></p>
<ol>
<li>The patient receives antibiotics for a sequence of four days, with gaps of one day allowed.</li>
<li>The sequence must start with a new antibiotic, defined as an antibiotic type that was not given in the previous two days.</li>
<li>The sequence must start within two days of a blood culture.</li>
<li>There must be at least one intervenous (I.V.) antibiotic within the +/-2 day window.</li>
</ol>
<p>Let's assess the first criterion by finding the first day of possible 4-day qualifying sequences.    </p>

In [29]:
# Create a new variable called day_of_first_new_abx_in_window
# .... YOUR CODE FOR TASK 7 ....


combinedDT[, day_of_first_new_abx_in_window := day_given[antibiotic_new == 1 & drug_in_bcx_window == 1][1], by = .(patient_id, blood_culture_day)]
head(combinedDT, 30)

# Remove rows where the day is before this first qualifying day
# .... YOUR CODE FOR TASK 7 ....

combinedDT <- combinedDT[day_given >= day_of_first_new_abx_in_window]
head(combinedDT)

patient_id day_given antibiotic_type route last_administration_day
1   1          2        ciprofloxacin   IV    NA                     
2   1          4        ciprofloxacin   IV     2                     
3   1          6        ciprofloxacin   IV     4                     
4   1          7        doxycycline     IV    NA                     
5   1          9        doxycycline     IV     7                     
6   1         15        penicillin      IV    NA                     
7   1         16        doxycycline     IV     9                     
8   1         18        ciprofloxacin   IV     6                     
9   1          2        ciprofloxacin   IV    NA                     
10  1          4        ciprofloxacin   IV     2                     
11  1          6        ciprofloxacin   IV     4                     
12  1          7        doxycycline     IV    NA                     
13  1          9        doxycycline     IV     7                     
14  1         15        penicillin      IV    NA                     
15  1         16        doxycycline     IV     9                     
16  1         18        ciprofloxacin   IV     6                     
17  8          1        doxycycline     PO    NA                     
18  8          2        penicillin      IV    NA                     
19  8          3        doxycycline     IV     1                     
20  8          6        doxycycline     PO     3                     
21  8          8        penicillin      PO     2                     
22  8         12        penicillin      IV     8                     
23  8          1        doxycycline     PO    NA                     
24  8          2        penicillin      IV    NA                     
25  8          3        doxycycline     IV     1                     
26  8          6        doxycycline     PO     3                     
27  8          8        penicillin      PO     2                     
28  8         12        penicillin      IV     8                     
29 23          1        doxycycline     IV    NA                     
30 23          1        penicillin      IV    NA                     
   days_since_last_admin antibiotic_new blood_culture_day drug_in_bcx_window
1  NA                    1               3                1                 
2   2                    0               3                1                 
3   2                    0               3                0                 
4  NA                    1               3                0                 
5   2                    0               3                0                 
6  NA                    1               3                0                 
7   7                    1               3                0                 
8  12                    1               3                0                 
9  NA                    1              13                0                 
10  2                    0              13                0                 
11  2                    0              13                0                 
12 NA                    1              13                0                 
13  2                    0              13                0                 
14 NA                    1              13                1                 
15  7                    1              13                0                 
16 12                    1              13                0                 
17 NA                    1               2                1                 
18 NA                    1               2                1                 
19  2                    0               2                1                 
20  3                    1               2                0                 
21  6                    1               2                0                 
22  4                    1               2                0                 
23 NA                    1              13                0   

patient_id day_given antibiotic_type route last_administration_day
1 1           2        ciprofloxacin   IV    NA                     
2 1           4        ciprofloxacin   IV     2                     
3 1           6        ciprofloxacin   IV     4                     
4 1           7        doxycycline     IV    NA                     
5 1           9        doxycycline     IV     7                     
6 1          15        penicillin      IV    NA                     
  days_since_last_admin antibiotic_new blood_culture_day drug_in_bcx_window
1 NA                    1              3                 1                 
2  2                    0              3                 1                 
3  2                    0              3                 0                 
4 NA                    1              3                 0                 
5  2                    0              3                 0                 
6 NA                    1              3                 0                 
  any_iv_in_bcx_window day_of_first_new_abx_in_window
1 1                    2                             
2 1                    2                             
3 1                    2                             
4 1                    2                             
5 1                    2                             
6 1                    2

## 8. Simplify the data
<p>The first criterion is: <em>The patient receives antibiotics for a sequence of four days, with gaps of one day allowed.</em></p>
<p>We've pinned down the first day of possible sequences in the previous task. Now we have to check for four-day sequences. We don't need the drug type (name); we need the days the antibiotics were administered.</p>

In [30]:
# Create a new data.table containing only patient_id, blood_culture_day, and day_given
simplified_data <- combinedDT[, .(patient_id, blood_culture_day, day_given)]

# Remove duplicate rows
#.... YOUR CODE FOR TASK 8 ....
simplified_data <- unique(simplified_data)
head(simplified_data)


patient_id blood_culture_day day_given
1 1          3                  2       
2 1          3                  4       
3 1          3                  6       
4 1          3                  7       
5 1          3                  9       
6 1          3                 15

## 9. Extract first four rows for each blood culture
<p>To check for four-day sequences, let's pull out the first four days (rows) for each patient/blood culture combination. Some patients will have less than four antibiotic days. We'll remove them first.</p>

In [34]:
# Count the antibiotic days within each patient/blood culture day combination
# .... YOUR CODE FOR TASK 9 ....

simplified_data[, num_antibiotic_days := .N, by = .(patient_id, blood_culture_day)]
#head(simplified_data, 10)

# Remove blood culture days with less than four rows 
simplified_data <- simplified_data[num_antibiotic_days >= 4]

# Select the first four days for each blood culture
first_four_days <- simplified_data[, .SD[1:4], by = .(patient_id, blood_culture_day)]
head(first_four_days, 10)

patient_id blood_culture_day day_given num_antibiotic_days
1   1         3                 2          8                 
2   1         3                 4          8                 
3   1         3                 6          8                 
4   1         3                 7          8                 
5   8         2                 1          6                 
6   8         2                 2          6                 
7   8         2                 3          6                 
8   8         2                 6          6                 
9  23         3                 1         13                 
10 23         3                 3         13

## 10. Consecutive sequence
<p>Now we need to check whether each four-day sequence qualifies by having no gaps of more than one day.</p>
<!--"Patient receives antibiotics for a sequence of 4 days, with gaps of 1 day allowed."-->

In [35]:
# Make the indicator for consecutive sequence
# .... YOUR CODE FOR TASK 9 ....
first_four_days[, four_in_seq := as.numeric(max(diff(day_given)) <= 2), by = .(patient_id, blood_culture_day)]
head(first_four_days, 10)

patient_id blood_culture_day day_given num_antibiotic_days four_in_seq
1   1         3                 2          8                  1          
2   1         3                 4          8                  1          
3   1         3                 6          8                  1          
4   1         3                 7          8                  1          
5   8         2                 1          6                  0          
6   8         2                 2          6                  0          
7   8         2                 3          6                  0          
8   8         2                 6          6                  0          
9  23         3                 1         13                  1          
10 23         3                 3         13                  1

## 11. Select the patients who meet criteria
<p>A patient would meet the criteria if any of their blood cultures were accompanied by a qualifying sequence of antibiotics. Now that we've determined which each blood culture qualify let's select the patients who meet the criteria.</p>

In [36]:
# Select the rows which have four_in_seq equal to 1
suspected_infection <- first_four_days[four_in_seq == 1]

# Retain only the patient_id column
# .... YOUR CODE FOR TASK 9 ....

suspected_infection <- suspected_infection[, .(patient_id)]

# Remove duplicates
# .... YOUR CODE FOR TASK 9 ....

suspected_infection <- unique(suspected_infection)

# Make an infection indicator
# .... YOUR CODE FOR TASK 9 ....
suspected_infection[, infection := 1]
head(suspected_infection)

patient_id infection
1   1        1        
2  23        1        
3  64        1        
4  76        1        
5 164        1        
6 176        1

## 12. Find the prevalence of sepsis
<p>In this project, we used two EHR datasets to flag patients who were suspected of having a serious infection. We also got a <code>data.table</code> workout!</p>
<p>So far, we've been looking at records of all antibiotics administered and blood cultures that occurred over two weeks at a particular hospital. However, not all patients who were hospitalized over this period are represented in <code>combinedDT</code> because not all of them took antibiotics or had blood culture tests. We have to read in and merge the rest of the patient information to see what percentage of patients at the hospital might have had a serious infection.</p>

In [40]:
# Read in "all_patients.csv"
all_patientsDT <- fread('all_patients.csv')

# Merge this with the infection flag data
all_patientsDT <- merge(all_patientsDT, suspected_infection, all = TRUE)

# Set any missing values of the infection flag to 0
# .... YOUR CODE FOR TASK 12 ....

all_patientsDT <- all_patientsDT[is.na(infection), infection := 0]

# Calculate the percentage of patients who met the criteria for presumed infection
ans  <- round((all_patientsDT[, mean(infection)] * 100), 2)
ans
#cat("The total patients meet the criteria for presumed infection is",ans,"%") 

[1] 14.94

### observation
14.94% of the patients meet the criteria for presumed infection.